src/event-handlers/badge-emails.js

```diff
@@ -0,0 +1,154 @@
+      .call()
+
+    const addressData = await badgeTCR.methods
+      .getAddressInfo(tokenAddress)
+      .call()
+    const request = await badgeTCR.methods
+      .getRequestInfo(tokenAddress, Number(addressData.numberOfRequests) - 1)
+      .call()
+
+    return [
+      {
+        account: request.parties[REQUESTER],
+        message: `Your request to ${
+          addressData.status === '1' ? 'add' : 'remove'
+        } the Ethfinex badge ${
+          addressData.status === '1' ? 'to' : 'from'
+        } the token with address ${tokenAddress} was challenged and awaits arbitration.`,
+        to: `/badge/${process.env.BADGE_ADDRESS}/${tokenAddress}`,
+        type: 'Dispute'
+      }
+    ]
+  },
+  WaitingOpponent: async (badgeTCR, event) => {
+    const addressData = await badgeTCR.methods
+      .getAddressInfo(event.returnValues._address)
+      .call()
+
+    return [
+      {
+        account: event.returnValues._party,
+        message: `The opponent funded his side of an appeal for the dispute on the ${
+          addressData.status === '1' ? 'addition' : 'removal'
+        } of the Ethfinex badge ${
+          addressData.status === '1' ? 'to' : 'from'
+        } the token with address ${
+          event.returnValues._address
+        }. You must fund your side of the appeal to not lose the case.`,
+        to: `/badge/${process.env.BADGE_ADDRESS}/${
+          event.returnValues._address
+        }`,
+        type: 'ShouldFund'
+      }
+    ]
+  },
+  NewPeriod: async (badgeTCR, event) => {
+    const APPEAL_PERIOD = '3'
+    if (event.returnValues._period !== APPEAL_PERIOD) return // Not appeal period.
+
+    const tokenAddress = await badgeTCR.methods
+      .disputeIDToAddress(event.returnValues._disputeID)
+      .call()
+    if (tokenAddress === ZERO_ADDRESS) return // Dispute is not related to Badge TCR.
+
+    const addressData = await badgeTCR.methods
+      .getAddressInfo(tokenAddress)
+      .call()
+    const request = await badgeTCR.methods
+      .getRequestInfo(tokenAddress, Number(addressData.numberOfRequests) - 1)
+      .call()
+    return request.parties
+      .filter(n => n.account !== ZERO_ADDRESS) // Parties array has 3 elements, the first of which is unused.
+      .map(party => ({
+        account: party,
+        message: `The arbitrator gave a ruling on the dispute over the request to ${
+          addressData.status === '1' ? 'add' : 'remove'
+        } the Ethfinex badge ${
+          addressData.status === '1' ? 'to' : 'from'
+        } the token with address ${tokenAddress}. The request entered the appeal period. Raise an appeal before the end of the appeal period if you think the ruling is incorrect.`,
+        to: `/badge/${process.env.BADGE_ADDRESS}/${tokenAddress}`,
+        type: 'RulingGiven'
+      }))
+  }
+}
+
+module.exports.post = async (_event, _context, callback) => {
+  const event =
+    typeof _event.body === 'string' ? JSON.parse(_event.body) : _event.body
```

Comment:
It's always a string right?

Reply:
If using `serverless invoke`, th data comes parsed. This supports both cases to make testing easier.

Reply:
https://serverless.com/framework/docs/providers/aws/cli-reference/invoke/

use `--raw`